src/TwitterDB.js

```diff
@@ -0,0 +1,462 @@
+      WHERE user_cache.screenname = $name;
+      `
+    ,{
+      $name: name
+    }
+    ,(err,row) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving profile: %s",err.Error);
+        reject(err);
+        return;
+      }
+      if(row !== undefined){
+        row.profile = JSON.parse(row.profile);
+        resolve(row);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+/**
+ * TwitterDB.prototype.set_twitter_profile - Insert/Update a Twitter profile
+ * into the database.
+ * @function
+ * @param  {type} id          Twitter ID of the profile.
+ * @param  {string} name      screenname of the profile.
+ * @param  {object} data      The profile data.
+ * @param  {number} timestamp The time when this data was *fetched*.
+ */
+TwitterDB.prototype.set_twitter_profile = function(id,name,data,timestamp){
+  this.db.run(
+    `
+    REPLACE INTO user_cache VALUES ($id,$name,$data,$timestamp);
+    `
+  ,{
+    $id: id,
+    $name: name,
+    $data: JSON.stringify(data),
+    $timestamp: timestamp
+  },
+  function (err) {
+    if(err){
+      log.error("TwitDB","Error storing profile: %s",err);
+      return;
+    }
+  });
+}
+
+TwitterDB.prototype.get_client_data = function(user_id){
+  return new Promise((resolve,reject) =>{
+    this.db.get(
+      `
+      SELECT *
+      FROM twitter_account
+      WHERE twitter_account.user_id = $id;
+      `
+    ,{
+      $id: user_id
+    }
+    ,(err,row) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving client data: %s",err.Error);
+        reject(err);
+      }
+      if(row !== undefined){
+        resolve(row);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+TwitterDB.prototype.get_matrixid_from_twitterid = function(twitter_id){
+return new Promise((resolve,reject) =>{
+  this.db.get(
+    `
+    SELECT user_id
+    FROM twitter_account
+    WHERE twitter_id = $id
+    `
+    ,{
+      $id:twitter_id
+    },(err,row) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving linked userid: %s",err.Error);
+        reject(err);
+      }
+      if(row !== undefined){
+        resolve(row.user_id);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+TwitterDB.prototype.get_linked_user_ids = function(id){
+  return new Promise((resolve,reject) =>{
+    this.db.all(
+      `
+      SELECT user_id
+      FROM twitter_account
+      `
+    ,(err,rows) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving linked ids: %s",err.Error);
+        reject(err);
+      }
+      if(rows !== undefined){
+        rows.forEach(function(val,i,arr){
+          arr[i] = val.user_id;
+        });
+        resolve(rows);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+TwitterDB.prototype.set_client_data = function(user_id,twitter_id,data){
+  this.db.run(
+    `
+    REPLACE INTO twitter_account VALUES (
+      $user_id,
```

Comment:
Be **very very careful** relying on column ordering... this is not a great idea if you ever want to modify the table... being explicit is always better.